### PR TITLE
Space group additions: `space_group_hall` and `space_group_it_number`

### PR DESCRIPTION
**Pull request by merkys**
Partially addresses #35  This PR introduces two symmetry space group notations, Hall symbol and IT number, closely following their usage in CIF. Both Hall symbol and IT number have their drawbacks: in some cases both need additional information to uniquely define symmetry space group of a crystal. To cover this drawback, a list of symmetry operations is needed, and its introduction is envisaged in #35. Nevertheless standardizing Hall symbol and IT number would facilitate searches on these widely used notations.

### CODE REVIEW

**Comment by rartino** [comment]
Correct me if I am wrong, but I have come to the conclusion that it is *not* 100% safe to assume there is a standard Hall symbol for all possibilities - in case a structure isn't placed with a "proper" origin. Is it absolutely safe to assume there is a non-standard Hall symbol that describes structures given with a completely arbitarily origin?

Also, I think I'm for MAY support level on these. If you don't know the symmetry information it can be very expensive (arguably sometimes even impossible) to assign them.

**Comment by merkys** [comment]
> Correct me if I am wrong, but I have come to the conclusion that it is _not_ 100% safe to assume there is a standard Hall symbol for all possibilities - in case a structure isn't placed with a "proper" origin. Is it absolutely safe to assume there is a non-standard Hall symbol that describes structures given with a completely arbitary origin?

Right, but maybe we can say that only standard Hall symbols are supported, and they should be provided only for structures with "proper" origin? My point is that if we cannot use standard Hall symbol/IT number to unambiguously define symmetry, these fields should be `null`, and in future symmetry operator lists should be used to define such situations.

> Also, I think I'm for MAY support level on these. If you don't know the symmetry information it can be very expensive (arguably sometimes even impossible) to assign them.

Sure, I am fine with OPTIONAL/MAY support level.

File: optimade.rst [hunk 1]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
...
+space\_group\_hall
+~~~~~~~~~~~~~~~~~~
+
+- **Description**: Symmetry space group name as defined in Hall, S. R. (1981), Acta Cryst. A37, 517-525 and erratum (1981), A37, 921.
+- **Type**: string
+- **Requirements/Conventions**:
+
+  - **Support**: SHOULD be supported by all implementations, i.e., SHOULD NOT be :val:`null`.
+  - **Query**: Support for queries on this property is OPTIONAL.
+  - Each component of the space group name MUST be separated by a single space symbol.
```
Comment (rartino) [inline optimade.rst:2068]:
```suggestion
  - Each component of the Hall symbol MUST be separated by a single space symbol.
  - If there exists a standard Hall symbol which represents the symmetry it SHOULD be used.
  - MUST be null if :property:`nperiodic_dimensions` is not equal to 3.
```

File: optimade.rst [hunk 2]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
 
   - :val:`[[4.0,0.0,0.0],[0.0,4.0,0.0],[0.0,1.0,4.0]]` represents a cell, where the first vector is :val:`(4, 0, 0)`, i.e., a vector aligned along the :val:`x` axis of length 4 Å; the second vector is :val:`(0, 4, 0)`; and the third vector is :val:`(0, 1, 4)`.
 
+space\_group\_hall
+~~~~~~~~~~~~~~~~~~
+
+- **Description**: Symmetry space group name as defined in Hall, S. R. (1981), Acta Cryst. A37, 517-525 and erratum (1981), A37, 921.
```
Comment (rartino) [inline optimade.rst:2062]: To be closer to the phrasing of the IUCr, and strictly expressing that it is not just any Hall symbol, but the one that represents the structure:
```suggestion
- **Description**: A Hall space group symbol representing the symmetry of the structure as defined in Hall, S. R. (1981), Acta Cryst. A37, 517-525 and erratum (1981), A37, 921.
```

File: optimade.rst [hunk 3]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
...
+
+- **Examples**:
+
+  - :val:`P 2c -2ac`
+  - :val:`-I 4bd 2ab 3`
+
+space\_group\_it\_number
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- **Description**: Symmetry space group number as defined in International Tables for Crystallography Vol. A.
```
Comment (rartino) [inline optimade.rst:2078]:
```suggestion
- **Description**: Space group number for the structure assigned by the International Tables for Crystallography Vol. A.
```

File: optimade.rst [hunk 4]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
...
+- **Type**: integer
+- **Requirements/Conventions**:
+
+  - **Support**: SHOULD be supported by all implementations, i.e., SHOULD NOT be :val:`null`.
+  - **Query**: Support for queries on this property is OPTIONAL.
+  - The integer value MUST be between 1 and 230.
```
Comment (rartino) [inline optimade.rst:2084]:
```suggestion
  - The integer value MUST be between 1 and 230.
  - MUST be null if :property:`nperiodic_dimensions` is not equal to 3.
```

**Review by rartino — CHANGES_REQUESTED** [review]
A few suggested changes

File: optimade.rst [hunk 5]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
...
   - :val:`[[4.0,0.0,0.0],[0.0,4.0,0.0],[0.0,1.0,4.0]]` represents a cell, where the first vector is :val:`(4, 0, 0)`, i.e., a vector aligned along the :val:`x` axis of length 4 Å; the second vector is :val:`(0, 4, 0)`; and the third vector is :val:`(0, 1, 4)`.
 
+space\_group\_hall
+~~~~~~~~~~~~~~~~~~
+
+- **Description**: Symmetry space group name as defined in Hall, S. R. (1981), Acta Cryst. A37, 517-525 and erratum (1981), A37, 921.
+- **Type**: string
+- **Requirements/Conventions**:
+
+  - **Support**: SHOULD be supported by all implementations, i.e., SHOULD NOT be :val:`null`.
```
Comment (rartino) [inline optimade.rst:2066]:
```suggestion
  - **Support**: OPTIONAL support in implementations, i.e., MAY be :val:`null`.
```

File: optimade.rst [hunk 6]
```diff
@@ -2056,6 +2056,33 @@ lattice\_vectors
...
+  - :val:`-I 4bd 2ab 3`
+
+space\_group\_it\_number
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- **Description**: Symmetry space group number as defined in International Tables for Crystallography Vol. A.
+- **Type**: integer
+- **Requirements/Conventions**:
+
+  - **Support**: SHOULD be supported by all implementations, i.e., SHOULD NOT be :val:`null`.
```
Comment (rartino) [inline optimade.rst:2084]:
```suggestion
  - **Support**: OPTIONAL support in implementations, i.e., MAY be :val:`null`.
```

**Review by ml-evs — APPROVED** [review]

**Pull request merged**